cwl/src/main/scala/cwl/CommandLineTool.scala

```diff
@@ -40,23 +43,55 @@ case class CommandLineTool private(
                                    temporaryFailCodes: Option[Array[Int]],
                                    permanentFailCodes: Option[Array[Int]]) {
 
-  def womExecutable(inputFile: Option[String] = None): Checked[Executable] =
-    CwlExecutableValidation.buildWomExecutable(ExecutableTaskDefinition.tryApply(taskDefinition).toEither, inputFile)
+  /** Builds an `Executable` directly from a `CommandLineTool` CWL with no parent workflow. */
+  def womExecutable(validator: RequirementsValidator, inputFile: Option[String] = None): Checked[Executable] = {
+    val taskDefinition = buildTaskDefinition(parentWorkflow = None, validator)
+    CwlExecutableValidation.buildWomExecutable(taskDefinition, inputFile)
+  }
+
+  private def validateRequirementsAndHints(parentWorkflow: Option[Workflow], validator: RequirementsValidator): ErrorOr[List[Requirement]] = {
+    import cats.instances.list._
+    import cats.syntax.traverse._
+
+    val allRequirements = requirements.toList.flatten ++ parentWorkflow.toList.flatMap(_.allRequirements)
+    // All requirements must validate or this fails.
+    val errorOrValidatedRequirements: ErrorOr[List[Requirement]] = allRequirements traverse validator
+
+    errorOrValidatedRequirements map { validRequirements =>
+      // Only Requirement hints, everything else is thrown out.
+      // TODO CWL don't throw them out but pass them back to the caller to do with as the caller pleases.
+      val hintRequirements = hints.toList.flatten.flatMap { _.select[Requirement] }
+      val parentHintRequirements = parentWorkflow.toList.flatMap(_.allHints)
+
+      // Throw out invalid Requirement hints.
+      // TODO CWL pass invalid hints back to the caller to do with as the caller pleases.
+      val validHints = (hintRequirements ++ parentHintRequirements).collect { case req if validator(req).isValid => req }
```

Comment:
aren't you revalidating them here ? I thought `val errorOrValidatedRequirements: ErrorOr[List[Requirement]] = allRequirements traverse validator` already validated them

Reply:
these are the `hints`